manifests/vcsscm/git.pp

```diff
@@ -45,6 +47,79 @@
           source => 'puppet:///modules/software/vcsscm/git/bash-git-prompt',
         }
       }
+
+      if $gitconfig or $gitignore {
```

Comment:
Seems like we only need this if `$gitconfig_user` or `$gitignore_user`.

I'm thinking of some other way to order this that may make it easier...